buildbot/osuosl/master/config/builders.py

```diff
@@ -3353,9 +3356,27 @@
                     "-DCMAKE_C_COMPILER_LAUNCHER=ccache",
                     "-DCMAKE_CXX_COMPILER_LAUNCHER=ccache"],
                 extra_stage2_cmake_args=[
-                    "-DLLVM_ENABLE_LLD=True",
-                    "-DCMAKE_C_FLAGS='-menable-experimental-extensions -march=rva23u64 -mllvm -force-tail-folding-style=data-with-evl -mllvm -prefer-predicate-over-epilogue=predicate-dont-vectorize'",
-                    "-DCMAKE_CXX_FLAGS='-menable-experimental-extensions -march=rva23u64 -mllvm -force-tail-folding-style=data-with-evl -mllvm -prefer-predicate-over-epilogue=predicate-dont-vectorize'"]
+                    util.Interpolate("-DLLVM_NATIVE_TOOL_DIR=%(prop:builddir)s/stage1.install/bin"),
+                    "-DLLVM_BUILD_TESTS=True",
+                    "-DPython3_EXECUTABLE=/usr/bin/python3",
+                    util.Interpolate("-DLLVM_EXTERNAL_LIT=%(prop:builddir)s/llvm-zorg/buildbot/riscv-rise/lit-on-qemu")],
+                stage2_toolchain_options=[
+                    "set(CMAKE_SYSTEM_NAME Linux)",
+                    "set(CMAKE_SYSROOT %(prop:builddir)s/../rvsysroot)",
+                    "set(CMAKE_C_COMPILER_TARGET riscv64-linux-gnu)",
+                    "set(CMAKE_CXX_COMPILER_TARGET riscv64-linux-gnu)",
+                    "set(CMAKE_C_FLAGS_INIT '-march=rva23u64 -mllvm -force-tail-folding-style=data-with-evl -mllvm -prefer-predicate-over-epilogue=predicate-else-scalar-epilogue')",
+                    "set(CMAKE_CXX_FLAGS_INIT '-march=rva23u64 -mllvm -force-tail-folding-style=data-with-evl -mllvm -prefer-predicate-over-epilogue=predicate-else-scalar-epilogue')",
+                    "set(CMAKE_LINKER_TYPE LLD)",
+                    "set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)",
+                    "set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)",
+                    "set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)",
+                    "set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)"],
+                env={
+                    "BB_IMG_DIR": util.Interpolate("%(prop:builddir)s/.."),
+                    "BB_QEMU_CPU": "zba=true,zbb=true,zbc=false,zbs=true,zfhmin=true,v=true,vext_spec=v1.0,zkt=true,zvfhmin=true,zvbb=true,zvkt=true,zihintntl=true,zicond=true,zcb=true,zfa=true,zawrs=true,rvv_ta_all_1s=true,rvv_ma_all_1s=true,rvv_vl_half_avl=true",
```

Comment:
Do we also need to enable zimop +zcmop? Can clang emit those currently?

Reply:
I don't believe we will, but I'll enable them for completeness - thanks. They weren't supported at the time I composed that command line, but they were added in qemu 9.1.0.

Thanks!

Reply:
A patch is under review for rva23 support in QEMU, so I'll additionally add a TODO here to switch over to that shorthand once it's merged and in a released qemu version https://lore.kernel.org/qemu-devel/20250115184316.2344583-1-dbarboza@ventanamicro.com/